presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -111,10 +111,27 @@ querySpecification
     : SELECT setQuantifier? selectItem (',' selectItem)*
       (FROM relation (',' relation)*)?
       (WHERE where=booleanExpression)?
-      (GROUP BY groupBy+=expression (',' groupBy+=expression)*)?
+      (groupBy=groupByClause)?
       (HAVING having=booleanExpression)?
     ;
 
+groupByClause
+    : GROUP BY groupingElement
+    ;
+
+groupingElement
+    : simpleGroupByExpressions+=expression (',' simpleGroupByExpressions+=expression)*    #simpleGroupByExpressions
+    | '(' ')'                                                                             #emptyGroupingSet
+    | ROLLUP rollupList=groupingColumnReferenceList                                       #rollupList
+    | CUBE cubeList=groupingColumnReferenceList                                           #cubeList
+    | GROUPING SETS groupingSets+=groupingColumnReferenceList (',' groupingSets+=groupingColumnReferenceList)* #groupingSet
+    ;
+
+groupingColumnReferenceList
```

Comment:
This doesn't look right. It would allow things like:

`GROUP BY ROLLUP a`
`GROUP BY CUBE a`
`GROUP BY GROUPING SETS a,b,c`